test/test_rgb2ycbcr.py

```diff
@@ -0,0 +1,182 @@
+                rgb.data_valid.next = True
+
+                for i in range(samples):
+
+                    # rgb signal assignment in the dut
+                    rgb.red.next = inputs['red'][i]
+                    rgb.green.next = inputs['green'][i]
+                    rgb.blue.next = inputs['blue'][i]
+
+                    if ycbcr.data_valid == 1:
+                        for ycbcr_act, val in zip(('y', 'cb', 'cr'),
+                                                  (int(ycbcr.y), int(ycbcr.cb),
+                                                   int(ycbcr.cr))):
+
+                            actual_outputs[ycbcr_act].append(val)
+
+                    yield clock.posedge
+
+                print_results(inputs, expected_outputs, actual_outputs)
+                raise StopSimulation
+
+            return tbdut, tbclk, tbstim
+
+        inst = bench_color_trans()
+        inst.config_sim(trace=True)
+        inst.run_sim()
+
+    else:
+
+        exp_y = tuple(expected_outputs['y'])
+        exp_cb = tuple(expected_outputs['cb'])
+        exp_cr = tuple(expected_outputs['cr'])
+
+        in_r = tuple(inputs['red'])
+        in_g = tuple(inputs['green'])
+        in_b = tuple(inputs['blue'])
+
+        y_s, cb_s, cr_s = [Signal(intbv(0)[pixel_bits:])
+                           for _ in range(3)]
+
+        @myhdl.block
+        def bench_color_trans():
+            tbdut = rgb2ycbcr(rgb, ycbcr, clock, reset, num_fractional_bits)
+            tbclk = clock_driver(clock)
+            tbrst = rstonstart(reset, clock)
+
+            @instance
+            def tbstim():
+                yield reset.negedge
+                rgb.data_valid.next = True
+
+                for i in range(samples):
+                    # rgb signal assignment in the dut
+                    rgb.red.next = in_r[i]
+                    rgb.green.next = in_g[i]
+                    rgb.blue.next = in_b[i]
+
+                    if ycbcr.data_valid == 1:
+
+                        # expected_outputs signal assignment
+                        y_s.next = exp_y[i-3]
+                        cb_s.next = exp_cb[i-3]
+                        cr_s.next = exp_cr[i-3]
+                        yield delay(1)
+                        print("Expected outputs ===>Y:%d Cb:%d Cr:%d"
+                              % (y_s, cb_s, cr_s))
+                        print("Actual outputs ===>Y:%d Cb:%d Cr:%d"
+                              % (ycbcr.y, ycbcr.cb, ycbcr.cr))
+                        print("----------------------------")
+                    yield clock.posedge
+
+                raise StopSimulation
+
+            return tbdut, tbclk, tbstim, tbrst
+
+        # verify and convert with iverilog
+        verify.simulator = 'iverilog'
+        assert bench_color_trans().verify_convert() == 0
+        # verify and convert with GHDL
+        verify.simulator = 'ghdl'
+        assert bench_color_trans().verify_convert() == 0
+
+
+def test_color_translation():
```

Comment:
Can you see the [results](https://coveralls.io/builds/6489970) of the coverage analysis?  It indicates the lines that were not tested.  You probably need to run the coverage manually (see the command in travis.yml) and look at the report to determine which 8 relevant lines were not tested.